src/node_group.cpp

```diff
@@ -20,22 +22,42 @@ void NodeGroup::remove_adjacent(NodeGroup* NG){
     }
 }
 
-const std::set<NodeGroup *> NodeGroup::get_adjacent() const{
+std::map<const NodeGroup *, OnePath> NodeGroup::get_adjacent() const{
+    return std::map<const NodeGroup *, OnePath>(adjacent.begin(), adjacent.end());
+}
+
+std::map<NodeGroup *, OnePath> NodeGroup::get_adjacent(){
```

Comment:
Why are there 2?

And why does the second one allow modification of the NodeGroups?

Reply:
One is constant, the other is not

`chunck.cpp 46
void Chunk::R_merge(NodeGroup &G1, std::set<NodeGroup*>& to_merge){
    if (to_merge.size() == 0){
        return;
    }
    std::set<NodeGroup*> new_merge;
    while(to_merge.size()>0){
        auto G2 = to_merge.begin();
        std::map<NodeGroup *, OnePath> to_add = G1.merge_groups(**(G2));
        delNodeGroup(**G2);
        new_merge.erase(*G2);
        ter->add_NodeGroup(&G1);
        for (std::pair<NodeGroup *const, OnePath> NG : to_add){
            if (contains_nodeGroup(NG.first) && &G1 != NG.first){
                new_merge.insert(NG.first);
            }
        }
        to_merge.erase(G2);
    }
    R_merge(G1, new_merge);
}
`
Here chunk modifies NodeGroups. (though Chunck "ownes" them so there might be another way around this.)